ts_scripts/sanity_utils.py

```diff
@@ -118,12 +118,18 @@ def run_grpc_test(model: dict):
     print("pass mg.mar_set=", mg.mar_set)
     mar_set_list_str = [str(s) for s in mg.mar_set]
     mar_set_str = ",".join(mar_set_list_str)
-    register_model_grpc_cmd = f"python ts_scripts/torchserve_grpc_client.py register {model_name} {mar_set_str}"
-    status = os.system(register_model_grpc_cmd)
-
-    if status != 0:
+    register_model_grpc_cmd = [
+        "python",
+        "ts_scripts/torchserve_grpc_client.py",
+        "register",
+        model_name,
+        mar_set_str
+    ]
+    result = subprocess.run(register_model_grpc_cmd, capture_output=True, text=True)
+
+    if result.returncode != 0:
         print("## Failed to register model with torchserve")
-        sys.exit(1)
+        return result.returncode
```

Comment:
This changes behavior from exiting program to returning with the error. We will need to look closer if it makes sense to continue.

Reply:
Fixed.